src/internal/connector/onedrive/collections.go

```diff
@@ -454,8 +480,12 @@ func (c *Collections) UpdateCollections(
 			// TODO(ashmrtn): Figure what when an item was moved (maybe) and add it to
 			// the exclude list.
 
-			col, found := c.CollectionMap[collectionPath.String()]
+			collectionID, ok := visitedPaths[collectionPath.String()]
```

Comment:
I think we actually do have the ID of the parent folder (along with the path) when looking at an item. I think it's at `item.GetParentReference().GetId()`. This is part of the `ParentReference` field that is returned which the API doc states is of type [itemReference](https://learn.microsoft.com/en-us/graph/api/resources/itemreference?view=graph-rest-1.0). That way the rest of the code can continue to stay the way it was and we'll just swap the map update to `c.CollectionMap[*item.GetParentReference().GetId()] = col`

I think we probably don't have the parent ID set in our tests right now because we've never needed it before :/ Graph Explorer shows it's populated for everything but the root folder item though (i.e. items in root have it set to root's ID, but root the folder item does not have it set -- see below)

Root folder parent reference:
```
"parentReference": {
    "driveType": "business",
    "driveId": "some-drive-ID"
},
```